src/android/GeofenceTransitionsIntentService.java

```diff
@@ -0,0 +1,265 @@
+        stackBuilder.addParentStack(MainActivity.class);
+
+        // Push the content Intent onto the stack.
+        stackBuilder.addNextIntent(notificationIntent);
+
+        // Get a PendingIntent containing the entire back stack.
+        PendingIntent notificationPendingIntent =
+                stackBuilder.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        // Get a notification builder that's compatible with platform versions >= 4
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this);
+
+        // Define the notification settings.
+        builder.setSmallIcon(R.drawable.icon)
+                // In a real app, you may want to use a library like Volley
+                // to decode the Bitmap.
+                .setLargeIcon(BitmapFactory.decodeResource(getResources(),
+                        R.drawable.icon))
+                .setColor(Color.RED)
+                .setContentTitle(notificationDetails)
+                .setContentText(getString(R.string.geofence_transition_notification_text))
+                .setContentIntent(notificationPendingIntent);
+
+        // Dismiss notification once the user touches it.
+        builder.setAutoCancel(true);
+
+        // Get an instance of the Notification manager
+        NotificationManager mNotificationManager =
+                (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+
+        // Issue the notification
+        mNotificationManager.notify(0, builder.build());
+    }
+
+    /**
+     * Maps geofence transition types to their human-readable equivalents.
+     *
+     * @param transitionType    A transition type constant defined in Geofence
+     * @return                  A String indicating the type of transition
+     */
+    private String getTransitionString(int transitionType) {
+        GPS_dbManager = new DBHelper(getApplicationContext(), "GPSList.db", null, 3);
+        switch (transitionType) {
+            case Geofence.GEOFENCE_TRANSITION_ENTER:
+                GPS_dbManager.insert("insert into SCAN_LIST values(null,'" + geofence.getRequestId() + "', '" + "Enter"+ "', '" + lat + "', '" + lon + "', '" + alt + "', '" + timestamp + "');");
+                return getString(R.string.geofence_transition_entered);
+            case Geofence.GEOFENCE_TRANSITION_EXIT:
+                GPS_dbManager.insert("insert into SCAN_LIST values(null,'" + geofence.getRequestId() + "', '" + "Exit"+ "', '" + lat + "', '" + lon + "', '" + alt + "', '" + timestamp + "');");
+                return getString(R.string.geofence_transition_exited);
+            default:
+                GPS_dbManager.insert("insert into SCAN_LIST values(null,'" + geofence.getRequestId() + "', '" + "UnKnow"+ "', '" + lat + "', '" + lon + "', '" + alt + "', '" + timestamp + "');");
+                return getString(R.string.unknown_geofence_transition);
+        }
+    }
+
+    @Override
```

Comment:
implementing the LocationListener interface on the IntentService for geo fencing will not work:

* http://stackoverflow.com/a/19979112
* http://stackoverflow.com/a/20212298
* http://stackoverflow.com/a/10321157

here's [one way to do it as a Service and separate LocationListener](http://stackoverflow.com/a/14478281)